### PR TITLE
Plans on JP: Analytics

### LINKED ISSUE #20692: Plans on JP: Analytics
See pc8eDl-Vv-p2

### PR DESCRIPTION
**Pull request by staskus**
Fixes #20692
pc8eDl-Vv-p2

- [ ] checkoutWebViewViewed not yet tracked, maybe could be done with the task that handles checkout?

### To test:

1. Log into free .com site and enable `Free to Paid plans dashboard card` feature flag 
2. `free_to_paid_plan_dashboard_card_shown` tracked when a dashboard card appears
3. `free_to_paid_plan_dashboard_card_tapped` tracked when a dashboard card tapped
4. `free_to_paid_plan_dashboard_card_menu_tapped` tracked when a dashboard card context menu tapped
5. `free_to_paid_plan_dashboard_card_menu_hidden` tracked when a dashboard card hidden
6. `domains_dashboard_select_domain_tapped` when domain selection is confirmed 
7.  `plan_selection_webview_viewed` when plan selection is presented

## Regression Notes
1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

None

8. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Free to Paid Plans/FreeToPaidPlansDashboardCardCell.swift
Comment (guarani) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Free to Paid Plans/FreeToPaidPlansDashboardCardCell.swift:38]: Not a blocker: 
It seems like we're not tracking this on [the "direct domains" dashboard card](https://github.com/wordpress-mobile/WordPress-iOS/blob/f972840fbfe672b4b1a52eaa84afccf10a9d80d3/WordPress/Classes/ViewRelated/Blog/Blog%20Dashboard/Cards/Domains/DashboardDomainsCardCell.swift#L22-L23) but that might be expected because it's not on [the list of analytics events](pc8eDl-NP-p2). Not sure of the context behind including it here but seems worthwhile to have the event if it can help in some way.

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Free to Paid Plans/PlansTracker.swift
Comment (guarani) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Free to Paid Plans/PlansTracker.swift]:
```suggestion
        WPAnalytics.track(.planSelectionWebViewViewed, properties: properties)
```

I think this is a typo

**Review by guarani — APPROVED** [review]
Thanks @staskus! 

Not a blocker: Do you think it would be useful to also add a `source` property to the [existing `domains_dashboard_select_domain_tapped` event](https://github.com/wordpress-mobile/WordPress-iOS/blob/416615c3313de86891b0e74b8fb2f97947d59707/WordPress/Classes/ViewRelated/Domains/Domain%20registration/RegisterDomainSuggestions/RegisterDomainSuggestionsViewController.swift#L221) so that we can attribute it correctly to this dashboard card?

Blocker: I think the typo mentioned below needs to be fixed, but I'll approve anyway so you can fix (assuming it is a typo) and merge without another review cycle.

> checkoutWebViewViewed not yet tracked, maybe could be done with the task that handles checkout?

Yep, I think it'll be easier to add afterwards.

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Free to Paid Plans/FreeToPaidPlansDashboardCardCell.swift
Comment (staskus) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Free to Paid Plans/FreeToPaidPlansDashboardCardCell.swift:38]: Yes, we weren't tracking it before but it was suggested we do it for this card.

**Comment by guarani** [comment]
> Not a blocker: Do you think it would be useful to also add a `source` property to the [existing `domains_dashboard_select_domain_tapped` event](https://github.com/wordpress-mobile/WordPress-iOS/blob/416615c3313de86891b0e74b8fb2f97947d59707/WordPress/Classes/ViewRelated/Domains/Domain%20registration/RegisterDomainSuggestions/RegisterDomainSuggestionsViewController.swift#L221) so that we can attribute it correctly to this dashboard card?

Just to close the loop: pc8eDl-Vv-p2#comment-769 (since it affects both platforms and this question is out-of-scope of this PR)

**Pull request merged**
